percona/controller/pgcluster/finalizer_test.go

```diff
@@ -0,0 +1,168 @@
+		By("Deleting the Namespace to perform the tests")
+		_ = k8sClient.Delete(ctx, namespace)
+	})
+
+	Context(v2beta1.FinalizerDeletePVC, Ordered, func() {
+		crName := ns + "-with-delete-pvc"
+		crNamespacedName := types.NamespacedName{Name: crName, Namespace: ns}
+		When("with finalizer", func() {
+			cr, err := readDefaultCR(crName, ns)
+			It("should read defautl cr.yaml", func() {
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			controllerutil.AddFinalizer(cr, v2beta1.FinalizerDeletePVC)
+
+			It("should create PerconaPGCluster", func() {
+				Expect(k8sClient.Create(ctx, cr)).Should(Succeed())
+			})
+
+			It("should create PVCs", func() {
+				_, err = reconciler().Reconcile(ctx, ctrl.Request{NamespacedName: crNamespacedName})
+				Expect(err).NotTo(HaveOccurred())
+				_, err = crunchyReconciler().Reconcile(ctx, ctrl.Request{NamespacedName: crNamespacedName})
+				Expect(err).NotTo(HaveOccurred())
+
+				pvcList := corev1.PersistentVolumeClaimList{}
+				Eventually(func() bool {
+					err := k8sClient.List(ctx, &pvcList, &client.ListOptions{
+						Namespace: cr.Namespace,
+						LabelSelector: labels.SelectorFromSet(map[string]string{
+							"postgres-operator.crunchydata.com/cluster": cr.Name,
+						}),
+					})
+					return err == nil
+				}, time.Second*15, time.Millisecond*250).Should(BeTrue())
+				Expect(len(pvcList.Items)).Should(Equal(4))
+			})
+
+			It("should delete PerconaPGCluster", func() {
+				Expect(k8sClient.Delete(ctx, cr)).Should(Succeed())
+			})
+
+			It("should run finalizer", func() {
+				_, err = reconciler().Reconcile(ctx, ctrl.Request{NamespacedName: crNamespacedName})
+				Expect(err).NotTo(HaveOccurred())
+				_, err = crunchyReconciler().Reconcile(ctx, ctrl.Request{NamespacedName: crNamespacedName})
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			It("should delete PVCs", func() {
+				pvcList := corev1.PersistentVolumeClaimList{}
+				Eventually(func() bool {
+					err := k8sClient.List(ctx, &pvcList, &client.ListOptions{
+						Namespace: cr.Namespace,
+						LabelSelector: labels.SelectorFromSet(map[string]string{
+							"postgres-operator.crunchydata.com/cluster": cr.Name,
+						}),
+					})
+					return err == nil
+				}, time.Second*15, time.Millisecond*250).Should(BeTrue())
+				for _, pvc := range pvcList.Items {
+					By(fmt.Sprintf("checking pvc/%s", pvc.Name))
+					Expect(pvc.DeletionTimestamp).ShouldNot(BeNil())
+				}
+			})
+		})
+
+		When("without finalizer", func() {
+			crName := ns + "-without-delete-pvc"
+			crNamespacedName := types.NamespacedName{Name: crName, Namespace: ns}
+
+			cr, err := readDefaultCR(crName, ns)
+			It("should read defautl cr.yaml", func() {
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			It("should create PerconaPGCluster", func() {
+				Expect(k8sClient.Create(ctx, cr)).Should(Succeed())
+			})
+
+			It("should reconcile PerconaPGCluster", func() {
+				_, err = reconciler().Reconcile(ctx, ctrl.Request{NamespacedName: crNamespacedName})
```

Comment:
You probably want to `Expect(err).NotTo(HaveOccurred())` after these reconcile calls.

Reply:
https://github.com/percona/percona-postgresql-operator/pull/405/commits/a7ef295f52c7d50b06d409a0fe94944aa34587ee